source/concurrent/ThreadPool.ooc

```diff
@@ -50,7 +50,10 @@ _Task: abstract class {
 			}
 			timer free()
 		}
-		this _state == _PromiseState Finished
+		this _mutex lock()
+		status = this _state == _PromiseState Finished
+		this _mutex unlock()
+		status
```

Comment:
Here is a good case to use `this _mutex with(|| status = this _state == _PromiseState Finished)` though, but doesn't matter to me.